perch/model.py

```diff
@@ -72,7 +72,11 @@ def __init__(self, **kwargs):
 
     def __getattr__(self, attr):
         try:
-            return self._resource[attr]
+            if self._resource.get(attr):
+                return self._resource[attr]
+            else:
+                defaults = self.schema(self._resource)
```

Comment:
Calling `self.schema` is not a good idea, because it has the side effect of raising a `voluptuous.Invalid` exception, which would be unexpected in `__getattr__`. For example, this should raise an `AttributeError` not a `voluptuous.Invalid` exception:

```python
org = perch.Organisation()
print org.name 
```

It should be possible to get `Required` keys in the `schema` and call the default function if there is one.

For it to work on `Organisation` and `User`, the schemas will need to be changed from class methods to class attributes.